data/tutorials/tt_02_file_manipulation.md

```diff
@@ -9,123 +9,618 @@ date: 2021-05-27T21:07:30-00:00
+
+```ocaml
+if Sys.file_exists "file.txt" then
+  print_endline "File exists."
+else
+  print_endline "File does not exist."
+```
+
+This code checks if `file.txt` exists and prints a message accordingly.
+
+### Checking If a Path is a Directory
+
+The `Sys.is_directory` function is used to check if a path points to a
+directory. It takes a path as an argument and returns a boolean value indicating
+whether the path is a directory.
+
+```ocaml
+if Sys.is_directory "/path/to/directory" then
+  print_endline "It is a directory."
+else
+  print_endline "It is not a directory."
+```
+
+This code checks if `/path/to/directory` is a directory and prints a message
+accordingly.
+
+### Changing File Permissions
+
+OCaml's `Unix` module provides a `chmod` function to change the permissions of a
+file. The function takes a filename and the new permissions (in Unix-style octal
+format) as arguments.
+
+```ocaml
+Unix.chmod "file.txt" 0o644
+```
+
+This code changes the permissions of `file.txt` to read and write for the owner
+and read-only for the group and others.
+
+### Getting File Size
+
+The `Unix.stat` function can be used to get the size of a file. This function
+returns a record with various file attributes, including its size.
+
+```ocaml
+let stats = Unix.stat "file.txt" in
+print_int stats.Unix.st_size
+```
+
+This code gets the size of `file.txt` and prints it.
+
+### Working with File Paths
+
+OCaml's `Filename` module provides several functions to work with file paths.
+
+```ocaml
+let dir = Filename.dirname "/path/to/file.txt" in
+let base = Filename.basename "/path/to/file.txt" in
+let full = Filename.concat dir "new_file.txt" in
+print_endline dir;
+print_endline base;
+print_endline full;
+```
+
+This code extracts the directory and base name from a path, and concatenates a
+directory name with a base name.
+
+### Creating Temporary Files and Directories
+
+The `Filename` module also provides functions to create temporary files and
+directories.
+
+```ocaml
+let temp_file = Filename.temp_file "prefix" ".suffix" in
+let temp_dir = Filename.temp_dir "prefix" in
+print_endline temp_file;
+print_endline temp_dir;
+```
+
+This code creates a temporary file and a temporary directory, and prints their
+paths.
+
+## Error Handling
+
+As with any operation involving I/O, file operations in OCaml can fail for a
+variety of reasons, such as when a file does not exist, a directory cannot be
+created due to insufficient permissions, or a disk is full. Handling such errors
+is crucial to writing robust and resilient programs. This section will cover
+some common error handling techniques in OCaml for file manipulation tasks.
+General purpose error handling in OCaml is also addressed in
+[dedicated tutorial](/docs/error-handling)
+
+### Catching Exceptions
+
+In OCaml, many of the file and directory operations can raise an exception when
+they encounter an error. For example, trying to open a non-existent file with
```

Comment:
"when they encounter an error" can be removed. Would it be useful to mention that all operations can raise `Sys_error` and not just `open_text` ?